drgn_tools/dentry.py

```diff
@@ -381,24 +381,46 @@ def __file_type(mode: Object) -> str:
     return "UNKN"
 
 
-def ls(prog: Program, directory: str, count: bool = False) -> None:
+def ls(
+    prog: Program,
+    directory: str,
+    parent_dentry: Optional[Object] = None,
+    depth: int = 0,
+    count: bool = False,
+) -> None:
     """
     Print dentry children, like the ls command
     :param directory: directory to print children of
+    :param parent_dentry: dentry object of the directory to print children of
```

Comment:
maybe mention that "if not provided, the dentry will be determined by looking up the directory by name"